rust/libnewsboat-ffi/src/lib.rs

```diff
@@ -35,6 +35,17 @@ pub extern "C" fn rs_replace_all(
     result.into_raw()
 }
 
+#[no_mangle]
+pub extern "C" fn rs_consolidate_whitespace( input: *const c_char) -> *mut c_char {
+    let rs_input = unsafe { CStr::from_ptr(input) };
+    let rs_input = rs_input.to_string_lossy().into_owned();
+
+    let result = utils::consolidate_whitespace(rs_input);
+
+    let result = CString::new(result).unwrap();
```

Comment:
`unwrap` may panic, so please add a comment explaining why it won't in this case.

(I'd request a comment for the `unsafe` above, too, but: 1) don't feel entitled to do this since I didn't do it in replace_all; 2) I think I'll be moving that char*-to-String conversion into a function of its own, and I'll comment it there.)

Reply:
Thanks.